test/pipe.civet

```diff
@@ -161,6 +178,22 @@ describe "pipe", ->
     foo((()=>{throw new Error})())
   """
 
+  testCase """
+    throw expression with thicc pipe
+    ---
+    x |> throw ||> y
+    ---
+    let ref;(y((ref = (()=>{throw x})())),ref)
+  """
+
+  testCase """
+    throw with placeholder
+    ---
+    x |> throw new Error .
+    ---
+    (()=>{return ($ => { throw new Error($)})})()(x)
```

Comment:
It might be nice to simplify this since we won't need to wrap with the IIFE if it is already a placeholder expression.

It was quite mindbending to figure out what was going on here.